mongo/cursor.go

```diff
@@ -124,7 +124,7 @@ func (c *Cursor) next(ctx context.Context, nonBlocking bool) bool {
 		// If we don't have a next batch
 		if !c.bc.Next(ctx) {
 			// Do we have an error? If so we return false.
-			c.err = c.bc.Err()
+			c.err = replaceErrors(c.bc.Err())
```

Comment:
I suppose this question applies to https://github.com/mongodb/mongo-go-driver/pull/506 as well, but if we are potentially replacing the error type returned, are we worried about breaking user code that could be checking the error type? Or does it not matter since users should be using `errors.Is` and we're wrapping the original error.